src/app/resolvers/test-result/test-result.resolver.ts

```diff
@@ -12,31 +12,12 @@ import { concatMap, map, mergeMap, Observable, of, take, tap } from 'rxjs';
 export class TestResultResolver implements Resolve<boolean> {
   constructor(private store: Store<State>, private action$: Actions) {}
 
-  /**
-   * Fetch test result if it is not already in the store and resolve to true if it is or fetch was successful.
-   * @param route
-   * @param state
-   * @returns true to resolve the route or false to block navigation
-   */
-  resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
-    return this.store.pipe(
-      select(selectedTestResultState),
+  resolve(): Observable<boolean> {
+    this.store.dispatch(fetchSelectedTestResult());
+    return this.action$.pipe(
+      ofType(fetchSelectedTestResultSuccess, fetchSelectedTestResultFailed),
       take(1),
-      concatMap((testResult) => of(!testResult ? true : false)),
-      tap((dispatch) => {
-        if (dispatch) {
-          this.store.dispatch(fetchSelectedTestResult());
-        }
-      }),
-      mergeMap((dispatch) => {
-        return dispatch
-          ? this.action$.pipe(
-              ofType(fetchSelectedTestResultSuccess, fetchSelectedTestResultFailed),
-              take(1),
-              map((action) => (action.type === fetchSelectedTestResultSuccess.type ? true : false))
-            )
-          : of(true);
-      })
+      map(action => (action.type === fetchSelectedTestResultSuccess.type ? true : false))
```

Comment:
```Suggestion
      map(action => action.type === fetchSelectedTestResultSuccess.type)
```

Reply:
Agree this is an equivalent truthy expression in my head

Reply:
Nice